dnf5/commands/download/download.cpp

```diff
@@ -56,13 +78,30 @@ void DownloadCommand::configure() {
     }
 
     context.update_repo_load_flags_from_specs(pkg_specs);
-    context.set_load_system_repo(false);
+    if (!alldeps_option->get_value()) {
+        context.set_load_system_repo(true);
+    } else {
+        context.set_load_system_repo(false);
+    }
     context.set_load_available_repos(Context::LoadAvailableRepos::ENABLED);
 }
 
+void DownloadCommand::goal_resolved() {
+    auto & transaction = *get_context().get_transaction();
+    auto transaction_problems = transaction.get_problems();
+    if (transaction_problems != libdnf::GoalProblem::NO_PROBLEM) {
+        if (transaction_problems != libdnf::GoalProblem::NOT_FOUND) {
+            throw GoalResolveError(transaction);
+        }
+    }
+}
+
 void DownloadCommand::run() {
     auto & ctx = get_context();
+    auto goal = ctx.get_goal();
```

Comment:
`auto goal = ctx.get_goal();` -> `auto goal = std::make_unique<libdnf::Goal>(ctx.base);`

We don't want to create a transaction in the context. Instead of `goal` in the context, we will create a local one. We only need local to resolve what to download.
And move this line to the condition `if (resolve_option->get_value()) {`. We don't need a `goal` elsewhere.
And also remove `goal->reset()`. After these changes, it will not be needed because the destructor will be called immediately.